lib/models/addon.js

```diff
@@ -1670,6 +1668,22 @@ Addon.prototype[BUILD_BABEL_OPTIONS_FOR_PREPROCESSORS] = function() {
   }
 };
 
+Object.defineProperty(Addon.prototype, 'nodeModulesPath', {
+  // This exists only for backward compat, because some addons use
+  // it. It was never marked public and we can totally nuke it, but we
+  // can be polite and give them a warning for a while.
+  get() {
+    if (!this._nodeModulesPath) {
+      let nodeModulesPath = require('node-modules-path');
+      let value = nodeModulesPath(this.root);
+      this._nodeModulesPath = { value };
+      this.ui.writeDeprecateLine('An addon is trying to access addon.nodeModulesPath. This is not a reliable way to discover npm modules. Instead, consider using require("resolve").sync(something, { basedir: addon.root })');
```

Comment:
We should include the stack here (or add an extra `debug` level) so folks have a chance at tracking this down

Reply:
We could also open issues for the addons we can see:

https://emberobserver.com/code-search?codeQuery=nodeModulesPath

Reply:
I added a stack-based message to identify the culprit.